Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -378,46 +431,43 @@
             *outExceptionString = toErrorString(nil, filesKey, @"at least one file must be specified");
             return false;
         }
+
+        parameters.files = makeVector<String>(files);
     }
 
-    if (!hasValidExecutionWorld(script, outExceptionString))
+    std::optional<WebExtensionContentWorldType> worldType;
+    if (!parseExecutionWorld(script, worldType, outExceptionString))
         return false;
 
-    return true;
-}
+    if (worldType)
+        parameters.world = worldType.value();
 
-bool WebExtensionAPIScripting::validateTarget(NSDictionary *targetInfo, NSString **outExceptionString)
-{
-    static NSArray<NSString *> *requiredKeys = @[
-        tabIDKey,
-    ];
-
-    static NSDictionary<NSString *, id> *keyTypes = @{
-        frameIDsKey: @[ NSNumber.class ],
-        tabIDKey: NSNumber.class,
-        allFramesKey: @YES.class,
-    };
-
-    if (!validateDictionary(targetInfo, targetKey, requiredKeys, keyTypes, outExceptionString))
-        return false;
+    if (JSValue *function = script[funcKey] ?: script[functionKey]) {
+        auto *key = script[funcKey] ? funcKey : functionKey;
```

Comment:
would be better to move this outside of the if statement so we could just do `JSValue *function = script[key]`